components/process.rst

```diff
@@ -128,6 +128,28 @@ environment variables using the second argument of the ``run()``,
     // On both Unix-like and Windows
     $process->run(null, ['MESSAGE' => 'Something to output']);
 
+Setting Environment Variables for Processes
+-------------------------------------------
+
+The constructor of the :class:`Symfony\\Component\\Process\\Process` class and
+all its methods related to executing processes (``run()``, ``mustRun()``,
+``start()``, etc.) allow passing an array of environment variables to set before
```

Comment:
I don't understand this. Env vars should be set before the process starts to execute, right? Otherwise, if you start setting the env vars at the same time that you start executing the process ... the env vars won't be available for the process, right?

Reply:
When you run a process, you pass several things at the same time with no order - they all have to be set to start the program: the command line, its pipes (stdin,stderr, etc), and its env vars.